tests/SideBySide/DataTypes.cs

```diff
@@ -495,6 +495,121 @@ public void QueryBinaryGuid(bool oldGuids)
 			}
 		}
 
+#if !BASELINE
+		[Theory]
+		[InlineData(MySqlGuidFormat.Default, false)]
+		[InlineData(MySqlGuidFormat.Default, true)]
+		[InlineData(MySqlGuidFormat.None, false)]
+		[InlineData(MySqlGuidFormat.Char36, false)]
+		[InlineData(MySqlGuidFormat.Char32, false)]
+		[InlineData(MySqlGuidFormat.Binary16, false)]
+		[InlineData(MySqlGuidFormat.TimeSwapBinary16, false)]
+		[InlineData(MySqlGuidFormat.LittleEndianBinary16, false)]
+		public void QueryGuidFormat(MySqlGuidFormat guidFormat, bool oldGuids)
+		{
+			bool isChar36 = guidFormat == MySqlGuidFormat.Char36 || (guidFormat == MySqlGuidFormat.Default && !oldGuids);
+			bool isChar32 = guidFormat == MySqlGuidFormat.Char32;
+			bool isBinary16 = guidFormat == MySqlGuidFormat.Binary16;
+			bool isTimeSwapBinary16 = guidFormat == MySqlGuidFormat.TimeSwapBinary16;
+			bool isLittleEndianBinary16 = guidFormat == MySqlGuidFormat.LittleEndianBinary16 || (guidFormat == MySqlGuidFormat.Default && oldGuids);
+
+			Guid guid = new Guid("00112233-4455-6677-8899-AABBCCDDEEFF");
+			string guidChar36 = "00112233-4455-6677-8899-AABBCCDDEEFF";
+			string guidChar32 = "00112233445566778899AABBCCDDEEFF";
+			byte[] guidBinary16 = { 0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, 0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF };
+			byte[] guidTimeSwapBinary16 = { 0x66, 0x77, 0x44, 0x55, 0x00, 0x11, 0x22, 0x33, 0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF };
+			byte[] guidLittleEndianBinary16 = { 0x33, 0x22, 0x11, 0x00, 0x55, 0x44, 0x77, 0x66, 0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF };
+
+			const bool mysql8 = false;
```

Comment:
Is there any way to detect MySQL 8+?

Reply:
I can think of several:

* Extend `ServerFeatures` enum with `UuidToBin` member, turn that off on other DBs
* Check `ServerVersion` at runtime for >= 8 (not ideal because MariaDB 10.3 may or may not support this function)
* Execute `SELECT UUID_TO_BIN` during test setup and set the flag based on whether an exception occurs

Overall I'd lean towards `ServerFeatures` as that's the standard way the test suite indicates whether a server feature can be expected to be present.

Reply:
Okay, that doesn't look too hard.